src/profile-logic/profile-store.js

```diff
@@ -24,36 +27,40 @@ export function uploadBinaryProfileData(): * {
         }
 
         xhr.onload = () => {
-          if (xhr.status === 200) {
-            resolve(xhr.responseText);
-          } else {
-            reject(
-              new Error(
-                `xhr onload with status != 200, xhr.statusText: ${
-                  xhr.statusText
-                }`
-              )
-            );
+          switch (xhr.status) {
+            case 413:
+              reject(
+                new Error(
+                  oneLine`
+                    The profile size is too large.
+                    You can try enabling some of the privacy features to trim its size down.
+                  `
+                )
+              );
+              break;
+            default:
+              if (xhr.status >= 200 && xhr.status <= 299) {
+                // Success!
+                resolve(xhr.responseText);
+              } else {
+                reject(
+                  new Error(
+                    `xhr onload with status != 200, xhr.statusText: ${
+                      xhr.statusText
+                    }`
+                  )
+                );
+              }
           }
         };
 
         xhr.onerror = () => {
           console.error(
-            'There was an XHR error in uploadBinaryProfileData()',
+            'There was an XHR network error in uploadBinaryProfileData()',
             xhr
           );
           reject(
-            new Error(
-              // The profile store does not give useful responses.
-              // See: https://github.com/firefox-devtools/profiler/issues/998
-              xhr.statusText
-                ? oneLine`
-                    There was an issue uploading the profile to the server. This is often
-                    caused by the profile file size being too large. The error
-                    response was: ${xhr.statusText}
-                  `
-                : 'Unable to make a connection to publish the profile.'
-            )
+            new Error('Unable to make a connection to publish the profile.')
```

Comment:
Should we still retain the `statusText` in case we get an error we are not expecting?

Reply:
This wasn't clear to me when we would receive an `error` event, the documentation isn't clear. I was under the impression that there would be no statusText, but you're right that we used to get this `error` event for big profiles... Does that depend on the HTTP status? I should try to upload a big file to the old server and see what happens for what... Thanks for the comment though, I think I'll add it back!